### PR TITLE
add Kubernetes ConfigMap Reload

### PR DESCRIPTION
**Pull request by cpanato**
* Image: add configmap-reload

Fixes: https://github.com/chainguard-images/images/issues/852

Related: 

### Pre-review Checklist

- [ ] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### Quality Requirements for Images PRs
The items in this checklist should all be checked in the PR with exceptions clearly documented.
The general idea is that to the extent possible, the image should be a drop-in replacement to its public counterpart.
- [ ] Image is a smaller size than its common public counterpart, or if not, the reason why should be explained
- [ ] Image should be scanned for CVEs and should return 0 or near-0 results. Exceptions should be explained.
- [x] The software in the image should be able to be installed and work as expected
  - [x] The image should start in a cluster
  - [x] The application should be accessible to the user/cluster/etc. as expected
- [ ] If there is an open source helm chart provided, the image should be able to be used with the chart. If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] build for x86_64 and aarch64 if possible, document exceptions

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [x] Include tests, sufficient enough that you would trust this image running in production.
- [ ] Perform manual tests as needed and document your testing results
  - [ ] For builder images (go, python, etc), build a sample app successfully
  - [ ] For services images (rabbit, databases, webservers) test basic functionality, upstream install/getting started, port availability, admin access. Document differences from public image.
  - [ ] Applications intended for use with a specific cloud provider must be tested on that cloud provider.

- [ ] The version included is the latest GA version of the software
- [x] The latest tag points to the newest stable version
- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
  - [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image contains the ca-certs bundle if needed
- [x] ENTRYPOINT
  - [x] For applications/servers/utilities call main program with no arguments e.g. [redis-server]

- [x] Add annotations e.g:
```
annotations:
  "org.opencontainers.image.authors": "Chainguard Team https://www.chainguard.dev/"
  "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/busybox/ # use the academy site here
  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/bazel # use github here
```
- [ ] Check if environment variables are needed e.g. to set data locations
- [ ] Ensure the image responds to SIGTERM
  - [ ] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [ ] Error logs write to stderr and normal logs to stdout. DO NOT write to file.
#### Documentation Requirements

- [x] Provide a README file, follow the [README template](readme-template.md) to include variants, tags, and usage examples
- [ ] Document your testing and results in the PR
- [ ] Document any exceptions to the testing and config requirements outlined in this template
- [ ] Document where and how the image deviates from popular alternatives (ports, packages, user, entrypoint, versions, etc). Explain why there's a difference and if the user needs to do anything differently when using the image

### CODE REVIEW

File: images/configmap-reload/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+docker pull cgr.dev/chainguard/configmap-reload:latest
+```
+
+## Usage
+
+This image is a drop-in replacement for the upstream image.
+You can run it using the helm chart with:
+
+```shell
+$ helm repo add autoscaler https://kubernetes.github.io/autoscaler
```
Comment (ajayk) [inline images/configmap-reload/README.md:34]: Autoscaler charts referred here

**Review by ajayk — CHANGES_REQUESTED** [review]

File: images/configmap-reload/README.md (hunk 1 above)
Comment (cpanato) [reply images/configmap-reload/README.md]: updated to use the one we are using in the tests, thanks for the review

File: images/configmap-reload/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+```shell
+$ helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+$ helm install my-release prometheus-community/alertmanager \
+    --set configmapReload.enabled=true \
+    --set configmapReload.image=cgr.dev/chainguard/configmap-reload \
+    --set image.tag=latest
+    <other configuration parameters here>
+```
+
+Note that the `configmap-reload` does need cloud provider configuration to work correctly, so it won't run locally.
```
Comment (ajayk) [inline images/configmap-reload/README.md:42]: still refers the autoscaler and doesn't need  cloud provider here

**Review by developer-guy — APPROVED** [review]

File: images/configmap-reload/README.md (hunk 2 above)
Comment (cpanato) [reply images/configmap-reload/README.md]: 🤦

**Review by ajayk — APPROVED** [review]

**Pull request merged**